lightning/src/chain/channelmonitor.rs

```diff
@@ -2299,6 +2321,39 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 	pub fn get_and_clear_pending_events(&mut self) -> Vec<Event> {
```

Comment:
I think we should update `ChainMonitor::process_pending_events` in consequence, noticing the user that this call is from now on important for the funds safety and call frequence should at least match every block ?